src/wgpu/wgpu-backend.cpp

```
@@ -0,0 +1,91 @@
+    }
+
+    WGPUInstance wgpuInstance = {};
+    SLANG_RETURN_ON_FAIL(createWGPUInstance(api, &wgpuInstance));
+    SLANG_RHI_DEFERRED({ api.wgpuInstanceRelease(wgpuInstance); });
+
+    WGPUAdapter wgpuAdapter = {};
+    SLANG_RETURN_ON_FAIL(createWGPUAdapter(api, wgpuInstance, &wgpuAdapter));
+    SLANG_RHI_DEFERRED({ api.wgpuAdapterRelease(wgpuAdapter); });
+
+    WGPUAdapterInfo wgpuAdapterInfo = {};
+    api.wgpuAdapterGetInfo(wgpuAdapter, &wgpuAdapterInfo);
```

Cppcheck (2.20.0)

[error] If memory allocation fails

(nullPointerArithmeticOutOfMemory)